clarify default value of requires_grad for tensors

dagitses opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#61038**
* #61037

Differential Revision: [D29491984](https://our.internmc.facebook.com/intern/diff/D29491984)

Commit:
clarify default value of requires_grad for tensors

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29490699).

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29491984).

albanD commented on docs/source/notes/autograd.rst:
```diff
@@ -91,6 +91,8 @@ the module level with :meth:`nn.Module.requires_grad_()`.
 When applied to a module, ``.requires_grad_()`` takes effect on all
 of the module's parameters (which have ``requires_grad=True`` by default).
 
+Tensors that are not module parameters have ``requires_grad=False` by default.
```
Maybe this should be just after the paragraph line 61 mentioning that all the Tensor factory Functions set it to False by default and it can be changed via the `requires_grad` keyword argument.
Also mention that when a Tensor is wrapped inside a `nn.Parameter`, then its `requires_grad` field is automatically set to True.

dagitses replied:
yeah i like that location better too. This is so far down I worry that no one would ever see it.

dagitses replied:
what do you think of this?

albanD commented on docs/source/notes/autograd.rst:
```diff
@@ -91,6 +91,8 @@ the module level with :meth:`nn.Module.requires_grad_()`.
 of the module's parameters (which have ``requires_grad=True`` by default).
 
+Tensors that are not module parameters have ``requires_grad=False` by default.
+
```
Note that the formatting above is missing the second backquote after the requires_grad

dagitses replied:
Thanks, but now moot.

BTW, I see some lines like :meth:`nn.Module.requires_grad_()`. Do you know why those are able to get away with single backquotes?

albanD replied:
The ones with a marker before like `:meth:` or `:attr:` need only single backquotes.
The inline code is marked by the double backquotes.

Commit:
Update on "clarify default value of requires_grad for tensors"


Differential Revision: [D29491984](https://our.internmc.facebook.com/intern/diff/D29491984)

[ghstack-poisoned]

Commit:
Update on "clarify default value of requires_grad for tensors"


Differential Revision: [D29491984](https://our.internmc.facebook.com/intern/diff/D29491984)

[ghstack-poisoned]

albanD reviewed (approved):
LGTM

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29491984).

Commit:
Update on "clarify default value of requires_grad for tensors"


Differential Revision: [D29491984](https://our.internmc.facebook.com/intern/diff/D29491984)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29491984).

Commit:
Update on "clarify default value of requires_grad for tensors"


Differential Revision: [D29491984](https://our.internmc.facebook.com/intern/diff/D29491984)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29491984).

Commit:
Update on "clarify default value of requires_grad for tensors"


Differential Revision: [D29491984](https://our.internmc.facebook.com/intern/diff/D29491984)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29491984).
